### ISSUE TITLE
Change description of CH0SGN in chipset reference

### ISSUE THREAD

**Issue by M3wP** [MEGA65/mega65-user-guide#278]
The current description is misleading.  It states:

> CH0SGN Enable Audio DMA channel 0 signed samples

The bit when set actually enables _unsigned_ samples.  The description should read:

> CH0SGN Enable Audio DMA channel 0 unsigned samples

**Comment by Impakt** [comment]
which file has this error?

**Comment by M3wP** [comment]
Oh sorry I thought I mentioned that.  The chipset reference manual.

**Comment by Impakt** [comment]
I can't find any reference to that in this repository.

**Comment by M3wP** [comment]
I have attached a section of a screenshot showing that it is there on page "L-22" of the M65 book.

![M65 Manual snippit](https://user-images.githubusercontent.com/10175607/133947342-afc2f671-3b05-4d71-867c-5cc9bc761460.png)

**Comment by johnwayner** [comment]
It looks like CH0SGN and CH1SGN were merged into one description for CHXSGN:
`• CHXSGN Enable Audio DMA channel X signed samples`
@M3wP, If you're sure it should be `unsigned` instead, I'll change it.

**Comment by M3wP** [comment]
Well...  Its been a while but I believe that setting the bit makes the samples unsigned so "Enable ... signed samples" is misleading.


Daniel.


________________________________

**Comment by johnwayner** [comment]
Well, it turns out that information is getting pulled from mega65-core's VHDL file for the processor...so I won't be changing that file.  

Looking at the VHDL for the DMA, the value of that bit is always XOR'd with the most significant bit of the sample (of 4, 8, or 16 bits).  This has the effect of rescaling the data from the signed range (say -8 to 7 for a 4-bit sample) to the unsigned range (0 to 15).  For example, -8 (1000 in 2's compliment) should turn to (0000), XORing the msb does that.  As it does for all 15 values.

So if the sample is signed and the CHXSGN bit is set that XOR will do that rescaling.  If the sample isn't signed, it'll just mess it up. Same if the sample is signed and CHXSGN isn't set.

All that to say, I think the documentation matches the implementation.  But maybe I'm wrong.  I'm not sure how you came to the conclusion that you did, so I can't consider that.  Needless to say, I guess I won't be changing anything at the moment.  Sorry.

**Comment by johnwayner** [comment]
It just dawned on me that the VHDL might had changed since you created this issue...and it has!  So I suspect maybe the behavior was faulty before and has now been fixed.  Rather than the documentation being wrong.

**Comment by dansanderson** [comment]
I believe the docstring is incorrect. I ran through some Audio DMA examples [in a recent Digest (March 2024)](https://dansanderson.com/mega65/justified-ancients/) and concluded that for CHXSGN, 0=signed and 1=unsigned. I distributed a demo disk with the issue that also follows this. This is the opposite of what is implied in the most recent docstring (mega65-core development branch): "Enable Audio DMA channel X signed samples"

I'm migrating this issue to mega65-core. (Our core PR process is not as fluid as for docs, so hopefully this will get it done by someone with admin access.) https://github.com/MEGA65/mega65-core/issues/807
